src/manifest.rs

```diff
@@ -125,105 +151,44 @@ impl RawManifest {
     }
 }
 
-mod serializer {
-    use super::*;
-    use serde::{ser::SerializeStruct, Serializer};
-
-    impl Serialize for RawManifest {
-        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-        where
-            S: Serializer,
-        {
-            match *self {
-                RawManifest::Canary {
-                    ref package,
-                    ref dependencies,
-                } => {
-                    let mut s = serializer.serialize_struct("Canary", 3)?;
-                    s.serialize_field("edition", CANARY_EDITION)?;
-                    s.serialize_field("package", package)?;
-                    s.serialize_field("dependencies", dependencies)?;
-                    s.end()
-                }
-                RawManifest::Unknown {
-                    ref package,
-                    ref dependencies,
-                } => {
-                    let mut s = serializer.serialize_struct("Unknown", 2)?;
-                    s.serialize_field("package", package)?;
-                    s.serialize_field("dependencies", dependencies)?;
-                    s.end()
-                }
-            }
+#[test]
+fn manifest() {
```

Comment:
Could you please add tests that:

1. Verify that the default edition is `CANARY` during serialization
2. Serialization fails when the edition doesnt match the format specified?

Also (nit) could you group them in a test mod?